src/app/CriticalSection.h

```diff
@@ -0,0 +1,27 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+namespace chip {
+namespace Platform {
+void CriticalSectionEnter();
```

Comment:
I would advocate the use of scoped locks for the needs of this PR.

Also, is there a reason why the lock lives outside of the `EventManagement` object?  That is the object that requires protection.  Why not just provide it the internal facilities it needs to protect itself?

If there is a good reason to introduce a new platform-namespace lock singleton, I think design of such a thing warrants additional scrutiny.

Reply:
Done, thanks